src/hooks/open-ai/useAppState/appStateProvider.tsx

```diff
@@ -0,0 +1,22 @@
+import { createContext, useState } from 'react'
+
+type ContextType = {
+	openAiToken: string
+	setOpenAiToken: React.Dispatch<React.SetStateAction<string>>
+}
+
+type ContainerProps = {
+	children: React.ReactNode
+}
+
+export const AppStateContext = createContext<ContextType>({} as ContextType)
```

Comment:
This change here seems hacky, but it allows us to avoid making `setOpenAiToken` nullable, when it will always have a value even at first render.

If we were using `const {stuff} = useContext(AppStateContext)` directly in our components, then it may be an issue, but since we wrap it up safely.

The default values are only read when you are trying to access the context without (or from outside of) a provider.

The way we implmented `AppStateProvider` and `useAppState`, this is not possible.

Reply:
The change being `createContext<ContextType>({} as ContextType)` instead of `createContext<ContextType>({
openAiToken: '',
setOpenAiToken: undefined,
})`